crates/ruff_python_parser/src/lexer.rs

```diff
@@ -523,6 +658,10 @@ impl<'source> Lexer<'source> {
             false
         };
 
+        if kind.is_any_fstring() {
+            return Ok(self.lex_fstring_start(quote, triple_quoted, kind.is_raw()));
+        }
+
```

Comment:
I'm thinking of removing the `StringKind::FString` and `StringKind::RawFString` as that is an invalid representation now that we'll be emitting different tokens for f-strings. Instead, I'm thinking of updating the `lex_identifier` to directly check for `f` (and related `F`, `r`, `R`) and call `lex_fstring_start` directly.

Reply:
This will be done after the linter changes are complete.